src/gfx/gfxutils.js

```diff
@@ -196,6 +196,25 @@ THREE.Raycaster.prototype.intersectVisibleObject = function (gfxObj, camera, cli
   return p;
 };
 
+THREE.Matrix4.prototype.extractScale = (function () {
+  const _v = new THREE.Vector3();
+
+  return function (scale) {
+    if (scale === undefined) scale = _v.clone();
```

Comment:
I think we should follow one of two ways here:
1. don't create new vector if `scale` is `undefined `and simply return 
2. create new vector for scale but at the same time log a warning with the fact, that `new` is too expensive operation to do it on-the-fly.
I would prefer the first one, what about you?

Reply:
Thanks, I didn't think about it